src/gentropy/dataset/l2g_feature_matrix.py

```diff
@@ -128,18 +130,32 @@ def calculate_feature_missingness_rate(
         }
 
     def fill_na(
-        self: L2GFeatureMatrix, value: float = 0.0, subset: list[str] | None = None
+        self: L2GFeatureMatrix, na_value: float = 0.0, subset: list[str] | None = None
     ) -> L2GFeatureMatrix:
         """Fill missing values in a column with a given value.
 
+        For features that correspond to gene attributes, missing values are imputed using the mean of the column.
+
         Args:
-            value (float): Value to replace missing values with. Defaults to 0.0.
+            na_value (float): Value to replace missing values with. Defaults to 0.0.
             subset (list[str] | None): Subset of columns to consider. Defaults to None.
 
         Returns:
             L2GFeatureMatrix: L2G feature matrix dataset
         """
-        self._df = self._df.fillna(value, subset=subset)
+        cols_to_impute = ["proteinGeneCount500kb", "geneCount500kb", "isProteinCoding"]
+        for col in cols_to_impute:
+            if col not in self._df.columns:
+                continue
+            else:
+                self._df = self._df.withColumn(
+                    col,
+                    f.when(
+                        f.col(col).isNull(),
+                        f.mean(f.col(col)).over(Window.partitionBy("geneId")),
```

Comment:
Why the window is by geneId? Shouldn't it be by studyLocusId?

Reply:
No. The studyLocus/gene ID pair will only have a null.